apps/frontend/src/views/Preview/Preview.tsx

```diff
@@ -1,45 +1,59 @@
+    const [render, setRender] = useState(null);
 
-    const handleElementClick = (element: PreviewElement) => {
+    //Make all the component unMount and re-mount
+    const handleValueUpdate = () => {
+        setShowComponent(false);
+        setTimeout(() => setShowComponent(true), 0);
+    };
+    const handleElementClick = (element: PreviewElementData) => {
         emitter.emit(`componentSelected`, element);
     };
+    useEffect(() => {
+        // TODO Fix this type error
+        setRender(
+            <div className={PreviewStyle.container}>
+                {previewTree.length > 0 &&
+                    previewTree.map((element, elementIndex) => {
+                        return (
+                            <div className={'tree-element'} key={`Element${elementIndex}`}>
+                                <PreviewComponentWrapper
+                                    index={elementIndex}
+                                    onClick={() => handleElementClick(element)}>
+                                    <Suspense>
+                                        <DynamicComponent
+                                            componentName={element.id}
+                                            customProps={element.props}
+                                        />
+                                    </Suspense>
+                                </PreviewComponentWrapper>
+                            </div>
+                        );
+                    })}
+                {previewTree.length == 0 && (
+                    <div className={'tree-element'} key={`Element-1}`}>
+                        <PreviewDroppable index={0} />
+                    </div>
+                )}
+            </div>
+        );
+        handleValueUpdate();
+    }, [previewTree]);
 
-    return (
-        <div className={PreviewStyle.container}>
-            {previewTree.length > 0 &&
-                previewTree.map((element, elementIndex) => {
-                    return (
-                        <div className={'tree-element'} key={`Element${elementIndex}`}>
-                            <PreviewComponentWrapper
-                                onClick={() => handleElementClick(element)}
-                                key={elementIndex}>
-                                <DynamicComponent
-                                    key={elementIndex}
-                                    componentName={element.id}
-                                    customProps={element.props}
-                                />
-                            </PreviewComponentWrapper>
-                        </div>
-                    );
-                })}
-            {previewTree.length == 0 && (
-                <div className={'tree-element'} key={`Element-1}`}>
-                    <PreviewDroppable />
-                </div>
-            )}
-        </div>
-    );
+    return <>{showComponent && render}</>;
```

Comment:
Didn't you get the behavior you want by just doing 
```jsx
    return <>{render}</>;
```
, as the `render` (renamed 👀 ) is already 'useEffected' ?